src/ResponsiveProvider.js

```diff
@@ -4,47 +4,40 @@ import PropTypes from 'prop-types';
 import React, { useMemo, useRef, useState } from 'react';
 import ResponsiveContext from './ResponsiveContext';
 
-const ResponsiveProvider = ({
-    initialMediaType,
-    defaultOrientation,
-    children,
-    breakpoints,
-    breakpointsMax,
-    mediaQueries,
-}) => {
-    const [isCalculated, setIsCalculated] = useState(false);
-    const [currentBreakpoint, setCurrentBreakpoint] = useState(
-        initialMediaType
-    );
+const useOrientation = (defaultOrientation) => {
     const [currentOrientation, setCurrentOrientation] = useState(
         defaultOrientation
     );
 
-    const breakpointsWithInitialValue = {
-        _initial: '0em',
-        ...breakpoints,
-    };
-
-    const breakpointNames = Object.keys(
-        mediaQueries || breakpointsWithInitialValue
-    );
+    // Add orientation media query
+    useEffect(() => {
+        const setMatchedOrientation = (matches) => {
+            setCurrentOrientation(matches ? 'portrait' : 'landscape');
+        };
 
-    if (!mediaQueries) {
-        mediaQueries = getMediaqueries(
-            breakpointsWithInitialValue,
-            breakpointsMax,
-            breakpointNames
+        const orientationMediaQueryList = window.matchMedia(
+            '(orientation: portrait)'
         );
-    }
 
-    const breakpointsRef = useRef(
-        getBreakpoints(mediaQueries, breakpointNames)
-    );
+        setMatchedOrientation(orientationMediaQueryList.matches);
 
-    const queriesObjects = useMemo(
-        () => getQueriesObjects(currentBreakpoint, breakpointsRef.current),
-        [currentBreakpoint, breakpointsRef]
-    );
+        const listener = (event) => {
+            setMatchedOrientation(event.matches);
+        };
+
+        orientationMediaQueryList.addListener(listener);
+
+        return () => {
+            orientationMediaQueryList.removeListener(listener);
+        };
+    }, []);
+
+    return currentOrientation;
+};
+
+const useMatchMedia = (breakpointsRef, initialMediaType) => {
```

Comment:
Now that I'm seeing, `useOrientation` also uses `matchMedia`, maybe `useMediaType` would be a more appropriate name...

Reply:
Changed from `useMatchMedia` to `useMediaType`.

Reply:
Yeap `useMediaType` seems better 👍